### ISSUE TITLE
`expect_unstable` lint for feature instability assertions

### ISSUE THREAD

**Issue by onkoe** [rust-lang/rust-clippy#13330]
### What it does

This lint notifies a user when a given feature is stable. It lets users mark blocks, items, and statements with an expectation: `#[expect_unstable("some_unstable_feature")]` 

## Configuration

- `follow-msrv`: Whether the crate's MSRV should be considered when checking for this lint. If the feature is stable, but the MSRV is set to something lower than the stabilized version, the lint will not apply. (default: `true`)

### Advantage

- Avoids confusing (and difficult-to-track) TODOs and FIXMEs where unstable features would improve source code.
- Allows users to get rid of [manual checks](https://github.com/onkoe/max6675-hal/blob/2f4c961e330c9c523828e6d9733d2b89f57a7dd0/.github/workflows/ci/main.rs) in CI.
- Refines project quality as Rust contributors refine the language.

### Example

```rust
// FIXME: make sure to use `core::error::Error` when stable!
// i sure hope someone notices this after 1.81..!
use std::error::Error;
```

Could be written as:

```rust
#[expect_unstable("error_in_core")]
use std::error::Error;
```

**Comment by Jarcho** [comment]
Just since I was confused when I read this the first time, this is intended to label code which is supposed to be changed once a feature becomes stable?

Nominated as this is related to a potential change to how we handle MSRVs.

**Comment by onkoe** [comment]
@Jarcho Yes, you have it right! It's intended to alert users when unstable features are stabilized.

**Comment by Jarcho** [comment]
Discussed this briefly at the last meeting. There were no objections to doing this. This won't need a config option for MSRV handling since #13416 would handle that in a more general way.

**Comment by Qix-** [comment]
Just to be clear, is this blocked on #13416? Is there any workaround for not having this lint right now?

I have a number of `core::convert::Infallible`s in code where the `!` would be the preferred spelling, whenever that feature is stable, for example.
